### PR TITLE
Task/tbq 28 refactoring to use command pattern

### PR DESCRIPTION
**Pull request by Sekator778**
this is the beginning of the commander from which we can all work in directions
Enjoy !

### CODE REVIEW

File: src/main/java/ua/shpp/eqbot/telegrambot/EqTelegramBot.java [hunk 1]
```diff
@@ -1,32 +1,90 @@
...
+                    if (execute) {
+                        isNewUser = false;
+                        userEntity = userRepository.findFirstById_telegram(telegram_id);
+                        users_active_session.put(telegram_id, convertToDto(userEntity));
+                    }
+                }
+                /*TODO  add logic here*/
+            } else if (!isNewUser || !messageText.equals("/reg")) {
+                /*TODO but if i don't want to register then what */
+                if (messageText.startsWith(COMMAND_PREFIX)) {
```
Comment (Kornilchenko) [inline src/main/java/ua/shpp/eqbot/telegrambot/EqTelegramBot.java]: I propose to replace
if (update.getMessage().isCommand()){

Comment (Kornilchenko) [reply src/main/java/ua/shpp/eqbot/telegrambot/EqTelegramBot.java]: properties specially from the project deleted?

File: pom.xml [hunk 2]
```diff
@@ -19,7 +19,7 @@
     <dependencies>
         <dependency>
             <groupId>org.springframework.boot</groupId>
-            <artifactId>spring-boot-starter</artifactId>
+            <artifactId>spring-boot-starter-web</artifactId>
```
Comment (Tsurkan) [inline pom.xml:22]: Why the ...web?

File: pom.xml [hunk 3]
```diff
@@ -36,8 +36,8 @@
             <artifactId>spring-boot-starter-data-jpa</artifactId>
         </dependency>
         <dependency>
-            <groupId>com.h2database</groupId>
```
Comment (Tsurkan) [inline pom.xml]: Do not delete H2 - it is for tests.

File: pom.xml [hunk 4]
```diff
@@ -50,6 +50,11 @@
             <artifactId>telegrambots-spring-boot-starter</artifactId>
             <version>6.1.0</version>
         </dependency>
+        <dependency>
```
Comment (Tsurkan) [inline pom.xml:58]: Why?

File: src/main/resources/application.properties [hunk 5]
```diff
@@ -1,28 +0,0 @@
-# system
```
Comment (Tsurkan) [inline src/main/resources/application.properties]: Add a settings file.

File: src/main/java/ua/shpp/eqbot/telegrambot/EqTelegramBot.java [hunk 6]
```diff
@@ -1,32 +1,90 @@
...
-    public void setProcessor(Processor processor) {
-        this.processor = processor;
-    }
-
-    private Processor processor;
-
     @Override
     public void onUpdateReceived(Update update) {
-        processor.process(update);
+        if (update.hasMessage() && update.getMessage().hasText()) {
```
Comment (Tsurkan) [inline src/main/java/ua/shpp/eqbot/telegrambot/EqTelegramBot.java]: Is it possible to somehow take the logic to a separate class?

File: src/main/java/ua/shpp/eqbot/command/StartCommand.java [hunk 7]
```diff
@@ -0,0 +1,22 @@
+package ua.shpp.eqbot.command;
+
+import org.telegram.telegrambots.meta.api.objects.Update;
+import ua.shpp.eqbot.service.SendBotMessageService;
+
+/**
+ * Start {@link Command}.
+ */
+public class StartCommand implements Command {
```
Comment (Tsurkan) [inline src/main/java/ua/shpp/eqbot/command/StartCommand.java:15]: Can I register again on the start menu? For example, if the city or name has changed.

File: src/main/java/ua/shpp/eqbot/command/SettingsCommand.java [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+
+    public static final String SETTINGS_MESSAGE = "Settings.\n";
+
+
+    public SettingsCommand(SendBotMessageService sendBotMessageService) {
+        this.sendBotMessageService = sendBotMessageService;
+    }
+
+    @Override
+    public boolean execute(Update update) {
```
Comment (Tsurkan) [inline src/main/java/ua/shpp/eqbot/command/SettingsCommand.java:23]: How to change the role to user?

File: src/main/java/ua/shpp/eqbot/telegrambot/EqTelegramBot.java (hunk 6 above)
Comment (Sekator778) [reply src/main/java/ua/shpp/eqbot/telegrambot/EqTelegramBot.java]: it goes beyond my task, it is being completed by Vitaly

File: src/main/java/ua/shpp/eqbot/command/StartCommand.java (hunk 7 above)
Comment (Sekator778) [reply src/main/java/ua/shpp/eqbot/command/StartCommand.java:15]: class start commander, its implementation is another task

File: src/main/java/ua/shpp/eqbot/command/SettingsCommand.java (hunk 8 above)
Comment (Sekator778) [reply src/main/java/ua/shpp/eqbot/command/SettingsCommand.java:23]: do we even need it?

**Comment by Sekator778** [comment]
recorded Mykhailo's remarks

**Review by Kornilchenko — APPROVED** [review]

File: src/main/java/ua/shpp/eqbot/telegrambot/EqTelegramBot.java (hunk 1 above)
Comment (Sekator778) [reply src/main/java/ua/shpp/eqbot/telegrambot/EqTelegramBot.java]: thank you, there was an error, the application.properties file needs to be corrected if you are concerned about it

**Review by Tsurkan — APPROVED** [review]

**Pull request merged**